presto-spi/src/main/java/com/facebook/presto/spi/type/VarcharType.java

```diff
@@ -28,6 +28,7 @@
 {
     public static final int MAX_LENGTH = Integer.MAX_VALUE;
     public static final VarcharType VARCHAR = new VarcharType(MAX_LENGTH);
+    public static final String VARCHAR_UNBOUNDED = "varchar(2147483647)";
```

Comment:
I wasn't aware of this discussion, but I think it's a way to go.

If I understand you correctly, todays commit should do exactly that.
I've removed possibility of creating unparametrized VARCHAR. This allows to remove support for this special case in many places.

This allowed to have a VARCHAR with only two special handling:
 1) Special case in parser to parse VARCHAR as VARCHAR(MAX_INT).
 2) Special case in toString in TypeSignature.
From user perspective it should be transparent.

Also, I've separated VarcharType.MAX_LENGTH and VarcharType.UNBOUNDED_LENGTH. VarcharType.UNBOUNDED_LENGTH should be private in future, but as there are some legacy cases this is left public. Same thing applies to getLength method which is marked @Deprecated.